packages/web3-utils/src/Utils.js

```diff
@@ -268,7 +268,7 @@ export const hexToNumber = (value) => {
         return value;
     }
 
-    return toBN(value).toNumber();
+    return parseInt(value, 16);
```

Comment:
Why is this necessary?

Reply:
To give proper result. This is a bug listed in #1677.